vespajlib/src/main/java/com/yahoo/concurrent/maintenance/Maintainer.java

```diff
@@ -92,17 +84,23 @@ public void awaitShutdown() {
     protected Duration interval() { return interval; }
 
     /** Run this while holding the job lock */
-    @SuppressWarnings("unused")
-    public final void lockAndMaintain() {
+    public final void lockAndMaintain(boolean force) {
+        if (!force && !jobControl.isActive(name())) return;
+        log.log(Level.FINE, () -> "Running " + this.getClass().getSimpleName());
+        jobMetrics.recordRunOf(name());
         try (var lock = jobControl.lockJob(name())) {
-            try {
-                jobMetrics.recordRunOf(name());
-                if (maintain()) jobMetrics.recordSuccessOf(name());
-            } finally {
-                // Always forward metrics
-                jobMetrics.forward(name());
+            if (maintain()) jobMetrics.recordSuccessOf(name());
+        } catch (UncheckedTimeoutException ignored) {
+            if (mode == Mode.shared) {
+                // This is fine as we're colliding with a run on another node
+                jobMetrics.recordSuccessOf(name());
```

Comment:
It's not a problem, but it's not really a success either.

Reply:
I think it would be better to change this so the `recordRunOf(name())` is called when the lock is acquired _or_ mode is `shared`, and record success only when it is an actual success. Otherwise a — somewhat contrived, perhaps — case where maintainers on all servers fail every time, but run long enough to block each other out, will seem like they all succeed every second time or so.

Reply:
When lock is acquired or mode is `exclusive` you mean?

Maybe we should remove special treatment of lock timeouts altogether instead. With staggered delay, it's unlikely that a run collides. If a maintainer frequently collides its interval is likely too low.

Reply:
That's what I mean :)

I agree with the reasoning about intervals.